generator/lib/src/code_chunks.dart

```diff
@@ -58,22 +61,46 @@ class CodeChunks {
   }
 
   static String propertyFieldName(ModelProperty property) {
+    if (property.type == OBXPropertyType.Relation) {
+      if (!property.name.endsWith('Id')) {
+        throw ArgumentError.value(property.name, 'property.name',
+            'Relation property name must end with "Id"');
+      }
+
+      return property.name.substring(0, property.name.length - 2);
+    }
+
     return property.name;
   }
 
   static String propertyReader(ModelProperty property) {
-    return "'${property.name}': inst.${propertyFieldName(property)}";
+    if (property.type == OBXPropertyType.Relation) {
+      return "'${property.name}': inst.${propertyFieldName(property)}.targetId";
+    } else {
+      return "'${property.name}': inst.${propertyFieldName(property)}";
+    }
   }
 
   static String propertyWriter(ModelProperty property) {
     final name = property.name;
+    final field = propertyFieldName(property);
     if (isTypedDataList(property)) {
-      return "r.${propertyFieldName(property)} = members['${name}'] == null ? null : ${property.dartFieldType}.fromList(members['${name}']);";
+      return "r.$field = members['${name}'] == null ? null : ${property
+          .dartFieldType}.fromList(members['${name}']);";
+    } else if (property.type == OBXPropertyType.Relation) {
+      return "r.$field.targetId = members['${name}'];" +
+          "\n r.$field.attach(store);";
```

Comment:
If there are multiple relations, the store is also attached multiple times (e.g. see `RelatedEntityB`). Currently no side effects, so I guess this is fine?

Reply:
It's attached to each relation itself so I also don't see an issue.

Reply:
🤦  Uh, sorry... Not an issue.